File: Java/942. DI String Match.java
class Solution {
    public int[] diStringMatch(String s) {
        int left = 0, right = s.length();
        int[] ans = new int[right+1];
        
        for (int i=0; i<s.length(); i++) {
            if (s.charAt(i) == 'I') {
                ans[i] = left++;
            } else {
                ans[i] = right--;
            }
        }
        
        ans[s.length()] = left;
        return ans;
    }
}

File: Python3/942. DI String Match.py
class Solution:
    def diStringMatch(self, s: str) -> List[int]:
        ans = []
        left, right = 0, len(s)
        
        for c in s: 
            if c == 'I':
                ans.append(left)
                left += 1
            else:
                ans.append(right)
                right -= 1
        
        ans.append(left)
        return ans
        
            